scapy/layers/bluetooth.py

```diff
@@ -477,7 +580,10 @@ class HCI_LE_Meta_Advertising_Report(Packet):
                     ByteEnumField("atype", 0, {0:"public", 1:"random"}),
                     LEMACField("addr", None),
                     FieldLenField("len", None, length_of="data", fmt="B"),
-                    StrLenField("data", "", length_from=lambda pkt:pkt.len), ]
+                    PacketListField("data", [], EIR_Hdr,
+                                    length_from=lambda pkt:pkt.len),
+                    SignedCharField("rssi", 0)]
```

Comment:
> Is the existing code wrong?

Yes, I'm pretty sure it was incorrect.

According to the BT 4.2 spec, Vol 2, Part E, 7.7.65.2 LE Advertising Report Event, a 1-byte long RSSI field follows the Data field, and the Length_Data field is as the name suggests, only the length of Data (excl. the RSSI field).